src/assets/htmx.md

```diff
@@ -0,0 +1,347 @@
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+</div>
+```
+
+After adding another
+
+```html
+<div id="todo-list">
+  <div>item 2</div>
+</div>
+```
+
+Instead, we want to only append the new todo item after the last todo item. so we use the attribute `hx-swap` to let htmx know where we want the response to render. the `beforeend` value will append the html response before the end of the targeted element
+
+After the swap attribute
+
+(adding item 1)
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+</div>
+```
+
+After adding another item (item 2)
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+  <div>item 2</div>
+</div>
+```
+
+
+
+
+But if we refresh the page we won't get the list of tasks we saved so we need to get the list
+For that, we need to make a get request
+
+```html
+<div id="todo-list" hx-get="/taskList" hx-trigger="load"></div>
+```
+
+Here we need to get the list when someone opens/loads the page so we give the trigger value of `load`. which will trigger when the page is load
+
+```go
+func listTask(w http.ResponseWriter, r *http.Request) {
+    tastList := getTODO() // gets data from database and returns html
+    io.WriteString(w, tastList)
+}
+func main() {
+    ...
+    http.HandleFunc("/save", listTask)
+    ...
+}
+```
+[demo](https://i.imgur.com/yMVvdP3.gif)
+
+Now the list is shown even when the page is refreshed.
+
+For edit let's try some thing different.
+
+First let's make clicking on the todo item go to a different page.
+For that, we need to get an anchor tag instead of a div with the todo text inside a div
+
+**example response from server**
+
+```htmx
+  <div hx-boost="true">
+      <a href="/task/asdasda">asdasda</a>
+  </div>
+```
+
+Here the `hx-boost` attribute will replace the body element and add the previous page to the history entry which will allow you to go back if needed.
+
+[demo](https://imgur.com/bktyFpm)
+
+For the actual editing, it will be similar to other examples.
+
+```html
+<div>
+  <div>asdasda</div>
+  <form id="myForm2" hx-put="/edit" hx-target="#app">
+    <input type="text" name="task" required="" />
+    <input type="hidden" name="old" value="asdasda" />
+    <input type="submit" value="edit" />
+  </form>
+</div>
+```
+
+Now let's perform deletion on the list.
+
+First, we have to make a delete button.
+
+For that, I will update the list template and add a delete form
+
+```html
+<div hx-boost="true" class="new-element">
+  <a href="/task/dasdas">dasdas</a>
+  <form
+    hx-trigger="click"
+    hx-delete="/delete"
+    hx-target="closest .new-element"
+    hx-swap="outerHTML swap:1s"
+  >
+    <input type="hidden" name="task" value="dasdas" />
+    delete
+  </form>
+</div>
+```
+
+Most of the snippets in the delete form are similar to other examples. but in the `hx-swap="outerHTML swap:1s"` attribute i added an extra value . All that value does is delay the swapping of the element by 1 second because i want to do a css animation during that time.
```

Comment:
- `the other examples`
- i => capital
- dot or comma?
- does the `value` do it or the key, or the parameter?